src/doctor.ts

```diff
@@ -0,0 +1,80 @@
+import { getProtocolInterface } from "./deps.ts";
+import { isNewSemverRelease } from "./utilities.ts";
+
+type RuntimeVersion = {
+  name: string;
+  current: string;
+} & RuntimeDetails;
+
+type RuntimeDetails = {
+  message?: string;
+  error?: {
+    message: string;
+  };
+};
+
+const getHostedDenoRuntimeVersion = async (): Promise<RuntimeDetails> => {
+  try {
+    const metadataURL = "https://api.slack.com/slackcli/metadata.json";
+    const response = await fetch(metadataURL);
```

Comment:
This can throw if the hostname can't be found. I hope that's rare because it doesn't seem obvious how we could catch this...

![Screenshot 2024-03-22 at 3 27 33 PM](https://github.com/slackapi/deno-slack-hooks/assets/18134219/da61e7a5-bcbe-4c16-b6d4-23aa93ebed3d)

Here's the stacktrace:

![Screenshot 2024-03-22 at 3 28 22 PM](https://github.com/slackapi/deno-slack-hooks/assets/18134219/79d08167-56f9-40bd-96be-487df6ca099c)